share/mk/bsd.cheri.mk

```diff
@@ -50,32 +50,8 @@ NEED_CHERI=	pure
 WANT_CHERI:= ${NEED_CHERI}
 .endif
 
-.if defined(LIB_CXX) || defined(PROG_CXX) || defined(SHLIB_CXX)
-.if ${MK_CHERI} != "no"
-# We need to use CHERI clang for C++ because we no longer build libstdc++
-WANT_CHERI?=	hybrid
-# XXXAR: leave this for a while until everyone has updated clang to
-# a version that defaults to libc++
-LDFLAGS+=	-stdlib=libc++
-.endif
-
-.endif
-
-.if ${MK_CHERI} != "no" && (!defined(WANT_CHERI) || ${WANT_CHERI} == "none")
-# When building MIPS code for CHERI ensure 16/32 byte stack alignment
-# for libraries because it could also be used by hybrid code
-# TODO: should be only for libraries and not programs
-.if ${COMPILER_TYPE} == "clang"
-# GCC doesn't support -mstack-alignment but I think it has been patched
-# to use 32 bytes anyway
-CFLAGS+=	-mstack-alignment=16
-.endif # $COMPILER_TYPE == clang
-.endif # MIPS, not hybrid (adjust stack alignment)
-
 .if ${MK_CHERI} != "no" && defined(WANT_CHERI) && ${WANT_CHERI} != "none"
-_CHERI_COMMON_FLAGS=	-integrated-as --target=mips64-unknown-freebsd \
-			-msoft-float \
-			-cheri-uintcap=${CHERI_UINTCAP_MODE:Uaddr}
+_CHERI_COMMON_FLAGS=	-cheri-uintcap=${CHERI_UINTCAP_MODE:Uaddr}
```

Comment:
I wonder if we can kill this.  The only user currently is usr.bin/truss/Makefile which sets it to 'addr' explicitly.

Reply:
Done.